ebrasca/fs Update

ebrasca opened the pull request:
Fix small errors in fat32 , cache
Don't display . and .. in directory
Refactor cache , disk

froggey commented:
I agree, block-device-read/write-sector look like they're just helper functions for the disk file systems. In fact, block-device-write-sector seems completely redundant and callers could be updated to directly call block-device-write.

ebrasca commented:
> The helper functions are specific to the supervisor disk block interface implementation, so I suggest moving them down past the API definition. Also, you have (flet ((read/write-sector ... which calls the (defun read/write-sector. This seems a bit confusing. Maybe use (flet ((%read/%write-sector instead so it's clearer?

Removed flet read/write-sector

> Also, changing (defun block-device-read/write-sector to (defmethod block-device-read/write-sector breaks all other block device implementations (that is, breaks USB). These two functions were added to support FAT32. Maybe they should be changed back to functions and moved to FAT32? They really shouldn't be a part of the block device API - I don't think the block device code should be allocating buffers for the user.

Are you sure it does break USB?
